rust/lance/src/index.rs

```diff
@@ -1019,9 +1028,25 @@ mod tests {
         assert_eq!(stats["num_unindexed_fragments"], 1);
         assert_eq!(stats["num_indices"], 1);
 
+        dataset
+            .optimize_indices(&OptimizeOptions {
+                num_indices_to_merge: 0,   // Just create index for delta
+                index_names: Some(vec![]), // Optimize nothing
+            })
+            .await
+            .unwrap();
```

Comment:
Do you want to add a test case where `index_names` is set but non-empty?

Reply:
good point, added tests to cover it